pkg/storage/utils/decomposedfs/tree/tree.go

```diff
@@ -794,12 +794,17 @@ func (t *Tree) Propagate(ctx context.Context, n *node.Node, sizeDiff int64) (err
 				}
 			case err != nil:
 				return err
+			case sizeDiff > 0:
+				newSize = treeSize + uint64(sizeDiff)
+			case uint64(-sizeDiff) > treeSize:
+				// The sizeDiff is larger than the current treesize. Which would result in
+				// a negative new treesize. Something must have gone wrong with the accounting.
+				// Reset the current treesize to 0.
+				sublog.Error().Uint64("treeSize", treeSize).Int64("sizeDiff", sizeDiff).
+					Msg("Error when updating treesize of parent node. Updated treesize < 0. Reestting to 0")
+				newSize = 0
 			default:
-				if sizeDiff > 0 {
-					newSize = treeSize + uint64(sizeDiff)
-				} else {
-					newSize = treeSize - uint64(-sizeDiff)
-				}
+				newSize = treeSize - uint64(-sizeDiff)
 			}
```

Comment:
I think this case statement makes the code harder to understand. I'd rather prefer to have it in the old if - else with an additional condition in the `sizeDiff < 0` branch.

Reply:
I dislike nested if/elses. I think new solution is cleaner in that regard. (just my opinion - not critical)

Reply:
Yeah, I also found the nested if inside the switch harder to read. Especially as I would need to add an additional if branch (or turn the nested if into a nested switch ...)

It might be more readable if I turn this into two separate switch commands though. Let me try ...

Reply:
> I'd rather prefer to have it in the old if - else with an additional condition in the `sizeDiff < 0` branch.

@dragotin Played around with this a bit. And adding another branch to the if statement will actually make the linter complain about the ifElseChain that should be turned into a switch statement ;-)

I guess I'll just leave it as is for now. (Splitting up the switch also doesn't help much with readability I think, as it will require an outer if condition. :man_shrugging: